.github/workflows/CD.yml

```
@@ -0,0 +1,166 @@
+        uses: actions/checkout@v4
+        with:
+          fetch-tags: true
+          fetch-depth: 0
+
+      - name: Get name
+        id: get-package-name
+        run: |
+          echo "package-name=$(yq '.project.name' pyproject.toml)" >> $GITHUB_OUTPUT
+        
+  release-conda-package:
+    name: Build with conda and release
+    runs-on: ubuntu-latest
+    needs: [check-release-tag, version-sanity-check, get-package-name]
+    if: ${{ needs.check-release-tag.outputs.is_release_tag == 'true' }}
+    env:
+      BUILD_FOLDER: ${{github.workspace}}/build
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4
+        with:
+            fetch-tags: true
+            fetch-depth: 0 
+      
+      - name: Set package path
+        id: set-package-path
+        run: |
+          echo "package-path=${{env.BUILD_FOLDER}}/noarch/*${{vars.PACKAGE_FORMAT}}" >> $GITHUB_OUTPUT
+      
+      - name: Setup conda build environment
+        uses: conda-incubator/setup-miniconda@a4260408e20b96e80095f42ff7f1a15b27dd94ca # v3.0.4
+        with:
+          miniconda-version: "latest"
+          python-version: ${{ vars.PY_VERSION }} 
+          environment-file: .conda/env_build.yml
+          auto-activate-base: false
+          auto-update-conda: false
+          show-channel-urls: true
+      
+      - name: Build conda package
+        shell: bash -el {0}
+        run: |
+          conda build . --no-anaconda-upload --output-folder=${{env.BUILD_FOLDER}} -c accessnri -c conda-forge -c coecms --package-format ${{vars.PACKAGE_FORMAT}}
+      
+      - name: Upload conda package to Anaconda.org
+        shell: bash -el {0}
+        run: |
+          anaconda -t ${{ secrets.ANACONDA_TOKEN }} upload --user ${{ secrets.ANACONDA_USER_NAME }} ${{steps.set-package-path.outputs.package-path}}
+      
+      - name: Create Release
+        uses: softprops/action-gh-release@c062e08bd532815e2082a85e87e3ef29c3e6d191 #v2.0.8
+        with:
+          tag_name: ${{ github.ref_name }}
+          name: ${{needs.get-package-name.outputs.package-name}} ${{ github.ref_name }}
+          generate_release_notes: true
+          fail_on_unmatched_files: true
+          files: |
+            ${{steps.set-package-path.outputs.package-path}}
+
+  cleanup-tag-on-failure:
+    name: Cleanup tag
+    needs: [check-release-tag, version-sanity-check, get-package-name, release-conda-package]
+    # Run this job if the tag is a release tag and any of the previous jobs failed (and don't skip if any of the needed jobs is skipped)
+    if: ${{ always() && failure() && needs.check-release-tag.outputs.is_release_tag == 'true' }}
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-tags: true
+
+      - name: Delete tag
+        run: |
+          git push origin :${{ github.ref_name }}
```

Bit of an edge case but this could delete a branch of the same name according to https://stackoverflow.com/a/5480292, so it might be worth specifying the tag namespace
```suggestion
          git push origin :refs/tags/${{ github.ref_name }}
```

---

Oh well, didn't even know you could have a branch named as a tag. 
It makes sense to be specific!

---

Changed to `git push origin :${{ github.ref }}` in [969ad5e922d252baad0d63ef975788d536caac94](https://github.com/ACCESS-NRI/umfile_utils/pull/26/commits/969ad5e922d252baad0d63ef975788d536caac94)